# Notebook for "Student Stress Analysis"
Amanda Curtis 

Started August 25th, 2025

In [2]:
# Importing packages

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import plotly.express as px
import seaborn as sns
sns.set_style("whitegrid")

## Table of Contents 
1. First Check and Cleaning 
2. Train / Test / Split
3. EDA #1
4. EDA #2
5. Feature Selection 

## First Check and Cleaning

For our first step, we import the data to the notebook. This data was obtained from Kaggle.com 

(link: https://www.kaggle.com/datasets/mdsultanulislamovi/student-stress-monitoring-datasets/data )

### Data Set 1

In [4]:
stress_1 = pd.read_csv("data/Stress_Dataset.csv")
stress_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 843 entries, 0 to 842
Data columns (total 26 columns):
 #   Column                                                                Non-Null Count  Dtype 
---  ------                                                                --------------  ----- 
 0   Gender                                                                843 non-null    int64 
 1   Age                                                                   843 non-null    int64 
 2   Have you recently experienced stress in your life?                    843 non-null    int64 
 3   Have you noticed a rapid heartbeat or palpitations?                   843 non-null    int64 
 4   Have you been dealing with anxiety or tension recently?               843 non-null    int64 
 5   Do you face any sleep problems or difficulties falling asleep?        843 non-null    int64 
 6   Have you been dealing with anxiety or tension recently?.1             843 non-null    int64 
 7   Have you

In [12]:
# Check for duplicates 
duplicate_rows = stress_1[stress_1.duplicated()]
print("All duplicate rows:")
print(duplicate_rows)

All duplicate rows:
     Gender  Age  Have you recently experienced stress in your life?  \
534       0   20                                                  5    
566       1   17                                                  3    
568       0   21                                                  4    
571       0   21                                                  3    
572       0   21                                                  3    
622       0   19                                                  1    
625       0   20                                                  2    
626       0   19                                                  2    
724       0   22                                                  2    
725       0   21                                                  1    
726       0   22                                                  4    
727       0   20                                                  1    
729       0   22                            

Notes on Stress_1: 

1. No missing values 
2. Integer values aside from last column (which is object)
3. 843 rows
4. 27 Dumplicated rows - what to do with those

### Data Set 2

In [5]:
stress_2 = pd.read_csv("data/StressLevelDataset.csv")
stress_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1100 entries, 0 to 1099
Data columns (total 21 columns):
 #   Column                        Non-Null Count  Dtype
---  ------                        --------------  -----
 0   anxiety_level                 1100 non-null   int64
 1   self_esteem                   1100 non-null   int64
 2   mental_health_history         1100 non-null   int64
 3   depression                    1100 non-null   int64
 4   headache                      1100 non-null   int64
 5   blood_pressure                1100 non-null   int64
 6   sleep_quality                 1100 non-null   int64
 7   breathing_problem             1100 non-null   int64
 8   noise_level                   1100 non-null   int64
 9   living_conditions             1100 non-null   int64
 10  safety                        1100 non-null   int64
 11  basic_needs                   1100 non-null   int64
 12  academic_performance          1100 non-null   int64
 13  study_load                    110

Notes on Stress_2: 

1. No missing values 
2. all integer values 
3. 1100 Rows

## Train / Test / Split

In [6]:
from sklearn.model_selection import train_test_split

In [ ]:
# Train